near-sdk/src/environment/hash.rs

```diff
@@ -36,9 +32,14 @@ impl CryptoHasher for Sha256 {
     fn hash(ingest: &[u8]) -> Self::Digest {
         unsafe { sys::sha256(ingest.len() as _, ingest.as_ptr() as _, ATOMIC_OP_REGISTER) };
 
-        let mut hash = [MaybeUninit::uninit(); 32];
-        read_register_fixed(ATOMIC_OP_REGISTER, &mut hash);
-        unsafe { std::mem::transmute(hash) }
+        let mut hash = [MaybeUninit::<u8>::uninit(); 32];
```

Comment:
Super ultra minor, but i'd move `let mut hash` inside `unsafe` block, as it's the part of overall "read 32 bytes" operation

Reply:
I generally try to minimize the amount of code inside of an `unsafe` block, but I'm fine with making this change. I might as well put the whole impl in the same block for readability if I do this.

Reply:
Ah, yeah, now I see why the code written the way it is!